Source/CarthageKit/Resolver.swift

```diff
@@ -280,6 +280,15 @@ private struct DependencyGraph: Equatable {
 			var nodeSet = edges[dependencyOf] ?? Set()
 			nodeSet.insert(node)
 			edges[dependencyOf] = nodeSet
+
+			// Add a nested dependency to the list of its ancestor.
+			let edgesCopy = edges
+			for (ancestor, var itsDependencies) in edgesCopy {
+				if itsDependencies.contains(dependencyOf) {
+					itsDependencies.insert(node)
+					edges[ancestor] = itsDependencies
+				}
+			}
```

Comment:
Should this also check to see if the nodes in `nodeSet` are in `edges`?

What happens if you call:

```
addNode(ReactiveCocoa, [ Box ])
addNode(ReactiveTask, [ ReactiveCocoa ])
```

?

Or do we know that these will always be ordered in the other direction?